### DISCUSSION TITLE
Ability to develop achievements offline

### DISCUSSION THREAD

**Discussion by Jamiras** [RetroAchievements/RAIntegration#770] [Ideas]
As a developer, I'd like to be able to create achievements without an active internet connection. Currently, to access any of the tools, I have to log in first.

* Minimally, allow updating and testing of Local achievements (no internet required)
* Ideally, allow adding/updating code notes (these would have to be queued and submitted once reconnected).
* Modification of non-Local achievements would not be expected.
* Earning of achievements while offline should be a separate issue if desired, and has its own set of concerns around possible cheating.

Suggest adding a new "Work Offline" menu option to enable the behavior. The user could just click past the "error connecting to retroachievements.org" dialog, or it could be modified to quickly enable the feature. In either case, once enabled, a dialog should be shown letting the user know they're in offline mode and that achievements cannot be earned until they reconnect.

**Comment by kdecks** [comment]
From #112 by @magusLOGS:

A Offline Mode would be nice for those of us who have a unstable internet connection or just want to work/play offline and then sync their progress manually to the server, similar to PSN/XBL.

Developing Offline
Making the entire developing progress offline compatible and allow to save a set/cheevos in one package-file, which can easily be transfered from one device to another device.

Earning Offline
A switch in the Emulators (like Hardcore) which turns on the 'Offline Mode'. Offline Mode would function similar to Softcore but it will require to Sync the earnings to users profile, all progress will be saved locally as well. Also being able to load the one package-file which contains every data for developing and earning cheevos incl. cheevo badges anytime a user want, with saving progress of which cheevo has been earned.

This way users can share their custom sets in form of a single file with other users without uploading their content to the Site.

**Comment by kdecks** [comment]
from @Jamiras, response:

Developing offline is already covered by #21, but it specifically states that earning achievements offline should be a separate issue, so please update this to reflect that.

The package-file concept is a new idea. The plan was just to leverage the existing files in RACache. Will have to think about that more.
